### PR TITLE
Support sampling in counted_terms aggregation

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we support using the `counted_terms` aggregation in a sampling context (e.g. as a sub-aggregation in a `random_sampler` aggregation).

### CODE REVIEW

File: x-pack/plugin/mapper-counted-keyword/src/yamlRestTest/resources/rest-api-spec/test/10_counted_terms.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+setup:
```
Comment (martijnvg) [inline x-pack/plugin/mapper-counted-keyword/src/yamlRestTest/resources/rest-api-spec/test/10_counted_terms.yml:1]: What is the reason of not adding these yaml tests to the `./x-pack/plugin/src/yamlRestTest/resources/test/counted_keyword` directory? In that case this PR change would be much smaller.

Comment (danielmitterdorfer) [reply x-pack/plugin/mapper-counted-keyword/src/yamlRestTest/resources/rest-api-spec/test/10_counted_terms.yml]: I honestly forgot that we had tests in the other directory. :man_facepalming: - I've moved the test in https://github.com/elastic/elasticsearch/pull/103846/commits/dbfa3505b312f21c364e53c4bb5736966de8065c.

Comment (martijnvg) [reply x-pack/plugin/mapper-counted-keyword/src/yamlRestTest/resources/rest-api-spec/test/10_counted_terms.yml]: No problem :) Sometimes we do keep the yaml tests in the module itself (e.g. aggregations module), with the reason to centralise all tests for a specific functionality. But then additional config changes need to be made in order to include these yaml tests in bwc tests.

**Review by martijnvg — APPROVED** [review]
LGTM

**Pull request merged**